homeassistant/components/tplink_omada/update.py

```diff
@@ -0,0 +1,149 @@
+    devices = await client.get_devices()
+    return [
+        FirmwareUpdateStatus(
+            device=d,
+            firmware=None
+            if not d.need_upgrade
+            else await client.get_firmware_details(d),
+        )
+        for d in devices
+    ]
+
+
+async def _poll_firmware_updates(
+    client: OmadaSiteClient,
+) -> dict[str, FirmwareUpdateStatus]:
+    """Poll the state of Omada Devices firmware update availability."""
+    return {d.device.mac: d for d in await _get_firmware_updates(client)}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switches."""
+    controller: OmadaSiteController = hass.data[DOMAIN][config_entry.entry_id]
+    omada_client = controller.omada_client
+
+    devices = await omada_client.get_devices()
+
+    coordinator = OmadaCoordinator[FirmwareUpdateStatus](
+        hass,
+        omada_client,
+        "Firmware Updates",
+        _poll_firmware_updates,
+        poll_delay=6 * 60 * 60,
+    )
+
+    entities: list = []
+    for device in devices:
+        entities.append(OmadaDeviceUpdate(coordinator, device))
+
+    async_add_entities(entities)
+    await coordinator.async_request_refresh()
+
+
+class OmadaDeviceUpdate(
+    OmadaDeviceEntity[FirmwareUpdateStatus],
+    UpdateEntity,
+):
+    """Firmware update status for Omada SDN devices."""
+
+    _attr_supported_features = (
+        UpdateEntityFeature.INSTALL
+        | UpdateEntityFeature.PROGRESS
+        | UpdateEntityFeature.RELEASE_NOTES
+    )
+    _firmware_update: OmadaFirmwareUpdate = None
+
+    def __init__(
+        self,
+        coordinator: OmadaCoordinator[FirmwareUpdateStatus],
+        device: OmadaListDevice,
+    ) -> None:
+        """Initialize the update entity."""
+        super().__init__(coordinator, device)
+
+        self._mac = device.mac
+        self._device = device
+        self._omada_client = coordinator.omada_client
+
+        self._attr_unique_id = f"{device.mac}_firmware"
+        self._attr_has_entity_name = True
+        self._attr_name = "Firmware Update"
+        self._refresh_state()
+
+    def _refresh_state(self) -> None:
+        if self._firmware_update and self._device.need_upgrade:
+            self._attr_installed_version = self._firmware_update.current_version
+            self._attr_latest_version = self._firmware_update.latest_version
+        else:
+            self._attr_installed_version = self._device.firmware_version
+            self._attr_latest_version = self._device.firmware_version
+        self._attr_in_progress = self._device.fw_download
+
+        if self._attr_in_progress:
+            # While firmware update is in progress, poll more frequently
+            async_call_later(self.hass, 60, self._request_refresh)
+
+    async def _request_refresh(self, _now: Any) -> None:
+        await self.coordinator.async_request_refresh()
```

Comment:
Proxy method, it add fuzz to the code, making it harder to follow. Instead, just use `self.coordinator.async_request_refresh()` in places used and remove this method.

Reply:
It's used in an attempt to schedule another refresh in 60 seconds during the firmware update. Is this an OK approach?
`self.coordinator.async_request_refresh` won't be passed directly into `async_call_later` without some encouragement.

Reply:
Is this any better?
```
        if self._attr_in_progress:
            # While firmware update is in progress, poll more frequently
            async def do_refresh(self, _now: Any) -> None:
                await self.coordinator.async_request_refresh()

            async_call_later(self.hass, 60, do_refresh)
```